### PR TITLE
Fix Core Foundation object leaks.

### PR DESCRIPTION
**Pull request by 0xZOne**
This fixes Core Foundation object leaks.

----
Unrelated, just as exciting: Today is the last day before the Chinese New Year, and I wish everyone a Happy New Year of the Dragon, Xin Nian Kuai Le! :)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: third_party/txt/src/txt/platform_mac.mm [hunk 1]
```diff
@@ -28,6 +28,16 @@
 namespace txt {
 
 const FourCharCode kWeightTag = 'wght';
+using CTFontPtr =
```
Comment (chinmaygarde) [inline third_party/txt/src/txt/platform_mac.mm:31]: Prefer using `fml::CFRef<T>` from `//flutter/fml/platform/darwin/cf_utils.h` instead.

File: third_party/txt/src/txt/platform_mac.mm [hunk 2]
```diff
@@ -43,39 +53,50 @@
 }
 
 CTFontRef MatchSystemUIFont(float desired_weight, float size) {
```
Comment (chinmaygarde) [inline third_party/txt/src/txt/platform_mac.mm:55]: The ownership of the CTFontRef isn't clear. Perhaps return fml::CFRef<CTFontRef> instead.

**Review by chinmaygarde — COMMENTED** [review]
Sound changes and lgtm but after using the utils already in fml. Thanks.

File: third_party/txt/src/txt/platform_mac.mm (hunk 1 above)
Comment (0xZOne) [reply third_party/txt/src/txt/platform_mac.mm]: done.

File: third_party/txt/src/txt/platform_mac.mm (hunk 2 above)
Comment (0xZOne) [reply third_party/txt/src/txt/platform_mac.mm]: done.

**Comment by 0xZOne** [comment]
PTAL

**Review by chinmaygarde — APPROVED** [review]
> Unrelated, just as exciting: Today is the last day before the Chinese New Year, and I wish everyone a Happy New Year of the Dragon, Xin Nian Kuai Le! :)

Happy New Year. Hope its happy and prosperous for you and yours! 🤗

**Review by jason-simmons — APPROVED** [review]

**Pull request merged**